File: examples/dialog_hello/dialog_hello.go
package main

import (
	"github.com/gen2brain/iup-go/iup"
)

func main() {
	iup.Open()
	defer iup.Close()

	fn := func(ih iup.Ihandle) int {
		return iup.CLOSE
	}

	dlg := iup.Dialog(
		iup.Vbox(
			iup.Label("Very Long Text Label").SetAttributes(`EXPAND=YES, ALIGNMENT=ACENTER`),
			iup.Button("Quit").SetHandle("quitBtName").SetCallback("ACTION", iup.ActionFunc(fn)),
		).SetAttributes(`ALIGNMENT=ACENTER, MARGIN=10x10, GAP=10`),
	).SetAttributes(`TITLE="Dialog", DEFAULTESC=quitBtName`)

	iup.Show(dlg)
	iup.MainLoop()
}

File: iup/external/examples/dialog_hello.c
#include <stdlib.h>
#include "iup.h"

static int quit_cb(Ihandle *self)
{
  (void)self;
  return IUP_CLOSE;
}

int main(int argc, char **argv)
{
  IupOpen(&argc, &argv);

  Ihandle *label = IupLabel("Very Long Text Label");
  IupSetAttribute(label, "EXPAND", "YES");
  IupSetAttribute(label, "ALIGNMENT", "ACENTER");

  Ihandle *button = IupButton("Quit", NULL);
  IupSetAttribute(button, "PADDING", "DEFAULTBUTTONPADDING");
  IupSetCallback(button, "ACTION", (Icallback)quit_cb);
  IupSetHandle("quitBtName", button);

  Ihandle *vbox = IupVbox(label, button, NULL);
  IupSetAttribute(vbox, "ALIGNMENT", "ACENTER");
  IupSetAttribute(vbox, "MARGIN", "10x10");
  IupSetAttribute(vbox, "GAP", "10");

  Ihandle *dlg = IupDialog(vbox);
  IupSetAttribute(dlg, "TITLE", "Dialog");
  IupSetAttribute(dlg, "DEFAULTESC", "quitBtName");

  IupShow(dlg);
  IupMainLoop();

  IupClose();
  return EXIT_SUCCESS;
}
